### PR TITLE
Context Menu 1: Basic scaffolding and simple actions

### PR DESCRIPTION
**Pull request by abey79**
### What

* Part of #4823

This PR is the first of a series to implement context menus in the Blueprint Tree. Here we setup the basic scaffolding and implement some basic actions:
- Add SV
- Add container
- Show/hide
- Remove

image

<br/>
<br/>

The selection is currently ignore and the actions only apply to the clicked item. TODO in future PRs:
- improved (multi-)selection handling
- additional actions
- entity level manipulations
- use ListItem instead of label/WidgetText
- better folder structure
- release checklist update

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5163/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5163/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5163/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5163)
- [Docs preview](https://rerun.io/preview/4b5e47083d3aa129352f0fbec9a74db500e20f16/docs) 
- [Examples preview](https://rerun.io/preview/4b5e47083d3aa129352f0fbec9a74db500e20f16/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Review by Wumpf — CHANGES_REQUESTED** [review]
The context menu allows you now to hide the top level container, something that isn't possible via a button and has no visual feedback.

More shockingly, you can also remove it and then not add a new one anymore 😱

**Comment by abey79** [comment]
> The context menu allows you now to hide the top level container, something that isn't possible via a button and has no visual feedback.
> 
> More shockingly, you can also remove it and then not add a new one anymore 😱

🤦🏻

File: crates/re_viewport/src/context_menu.rs [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+use crate::{Contents, ViewportBlueprint};
+use itertools::Itertools;
+use re_log_types::{EntityPath, EntityPathFilter};
+use re_space_view::{DataQueryBlueprint, SpaceViewBlueprint};
+use re_viewer_context::{ContainerId, Item, SpaceViewClassIdentifier, ViewerContext};
+
+/// Trait for things that can populate a context menu
+trait ContextMenuItem {
+    //TODO(ab): should probably return `egui::WidgetText` instead
```
Comment (Wumpf) [inline crates/re_viewport/src/context_menu.rs:11]: so.. why not? :)

File: crates/re_viewport/src/context_menu.rs [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+    fn label(&self, _ctx: &ViewerContext<'_>, _viewport_blueprint: &ViewportBlueprint) -> String {
```
Comment (Wumpf) [inline crates/re_viewport/src/context_menu.rs:12]: Just a label is a bit little though? Should add icons in a follow-up pr so we can show the familiar space view and layout icons - they are much faster to parse than text

File: crates/re_viewport/src/context_menu.rs (hunk 1 above)
Comment (abey79) [reply crates/re_viewport/src/context_menu.rs:11]: That comment is actually wrong. The end-game is to return a `ListItem` (with all the fanciness that comes with it: nicer look, icon, etc.). Both `ListItem` and the egui context menu code needs a bit of a overhaul for this to work though, so it'll come as a follow-up.

File: crates/re_viewport/src/context_menu.rs (hunk 2 above)
Comment (abey79) [reply crates/re_viewport/src/context_menu.rs:12]: Yup—see my other comment.

**Pull request merged**